### PR TITLE
Compare different AGIPD geometries

### PR DESCRIPTION
**Pull request by takluyver**
This work-in-progress branch adds a method for graphically comparing different geometries, so you can see how they've changed.

example comparison plot

### CODE REVIEW

**Comment by takluyver** [comment]
I think this is ready for review.

**Comment by takluyver** [comment]
Ping @tmichela @ebadkamil - this is only about 100 lines of new code, and it also fixes a dependency problem from cfelpyutils that is making the tests fail on Travis.

File: karabo_data/geometry2.py [hunk 1]
```diff
@@ -147,7 +150,7 @@ def from_crystfel_geom(cls, filename):
             for a in range(8):
                 d = geom_dict['panels']['p{}a{}'.format(p, a)]
                 tiles.append(AGIPDGeometryFragment.from_panel_dict(d))
-        return cls(modules)
+        return cls(modules,  filename=filename)
```
Comment (tmichela) [inline karabo_data/geometry2.py]: One extra space character

**Comment by tmichela** [comment]
LGTM, thanks

File: karabo_data/geometry2.py [hunk 2]
```diff
@@ -99,8 +99,11 @@ class AGIPD_1MGeometry:
     of the pixel size.
     """
     pixel_size = 2e-7  # 2e-7 metres == 0.2 mm
```
Comment (ebadkamil) [inline karabo_data/geometry2.py]: pixel_size = 2e-4

File: karabo_data/geometry2.py (hunk 1 above)
Comment (takluyver) [reply karabo_data/geometry2.py]: Eagle eyes :-P

Fixed

File: karabo_data/geometry2.py (hunk 2 above)
Comment (takluyver) [reply karabo_data/geometry2.py]: Yup, but you're already on that in PR #97 :+1:

**Pull request merged**